packages/go_router/test/delegate_test.dart

```diff
@@ -54,6 +54,27 @@ void main() {
     });
   });
 
+  group('popUntil', () {
+    testWidgets('It should pop 2 pages', (WidgetTester tester) async {
+      final GoRouter goRouter = await createGoRouter(tester)
+        ..push('/a')
+        ..push('/a');
+
+      await tester.pumpAndSettle();
+      expect(goRouter.routerDelegate.matches.matches.length, 3);
+      goRouter.routerDelegate.addListener(expectAsync0(count: 2, () {}));
```

Comment:
With the current implementation (simply calling `pop`, the notifiers are notified for each `pop`). In this case, 2 times. Is it correct or should the router delegate notify only once?

Reply:
we may need to refactor the code a bit, every time it notify the listener, the Gorouter will also notify its listener about location change, so if url = '/a/b/c/d/e' and popUtil to '/a'
if we notify each time it pop
goRoute will notify its listener with
'/a/b/c/d'
'/a/b/c'
'/a/b'
'/a'

instead just 
'/a'

Reply:
So, to be correct, this test should pass with `count: 1`, is that correct?

Reply:
yes

Reply:
Hello @chunhtai , I'm not sure it will be possible to notify the listeners only once even when popping several times (and make the test pass with `count: 1`). 

I pushed some code in https://github.com/flutter/packages/pull/2728/commits/2c48f397224c395df82d897eade32ee8077ba101 . Itis not correct as I guess it doesn't work with multiple nested navigators, it's more of a demonstration example. it uses directly `popUntil` from the `NavigatorState` and it still notifies the listeners twice.

Or would you know a way to make this work?